File: QKcOCkxgO.cpp
#include <iostream>

using namespace std;

int isPrime(int x){
   int divs = 1;

   for (int i = 1; i < x; i++){
      if (x % i == 0){
         divs++;
      }

      // not a prime number
      if (divs > 2){
         return -1;
      }
   }
return 1;
}

int main() {

   int input;

   cin >> input;

   if (isPrime(input) > 0){
      if (isPrime(input+2) > 0){
         cout << "PRIMO CASADO!";
      } else {
         cout << "PRIMO!";
      }
   } else {
      cout << "NAO PRIMO!";
   }
   return 0;
}

File: haskell/QKcOCkxgO.hs
isPrime :: Int -> Bool
isPrime num = if null [ x | x <- [2..(num-1)], num `mod` x == 0] then True else False

isCousin :: Int -> String
isCousin num =
  if (isPrime num)
    then if (isPrime num && isPrime (num+2))
      then "PRIMO CASADO!"
        else "PRIMO!"
    else "NAO PRIMO!"


main = do
  x <- getLine
  print (isCousin (read x))
